provider/aws/aws.go

```diff
@@ -509,9 +528,30 @@ func (p *AWSProvider) submitChanges(ctx context.Context, changes []*route53.Chan
 	for z, cs := range changesByZone {
 		var failedUpdate bool
 
-		batchCs := batchChangeSet(cs, p.batchChangeSize)
+		// group changes into new changes and into changes that failed in a previous iteration and are retried
+		var newChanges, retriedChanges Route53Changes
+		for _, c := range cs {
+			found := false
+			if p.failedChangesQueue[z] != nil {
+				for _, failedChange := range p.failedChangesQueue[z] {
+					if c == failedChange {
+						retriedChanges = append(retriedChanges, c)
+						found = true
```

Comment:
Do we maybe want to add a `break` here? It seems to be a common pattern. If `c` was found in `failedChangesQueue` we don't need to keep looking. Also, if for some reason it's found multiple times in `failedChangesQueue` it would be added to `retriedChanges` multiple times as well. Not sure if that's what you want or not, though.

Reply:
It would be helpful to extract these lines into a function `findChangeInQueue(c, p.failedChangesQueue[z])` that takes the item `c` and the queue where you look for it. That's essentially all that's happening on lines 534 to 541.

Reply:
Moved out the whole chunk into `findChangesInQueue`.